### DISCUSSION TITLE
If you can read me, it means that you misconfigured Lychee.

### DISCUSSION THREAD

**Discussion by sloniupl** [LycheeOrg/Lychee#2424] [Q&A] [answered]
I started receiving below notification on one of my galleries recently:

"If you can read me, it means that you misconfigured Lychee.
Please check that:

your APP_URL is properly set to the correct url.
For example: APP_URL=https://lychee.example.com
if you are working behind a reverse proxy, that TRUSTED_PROXIES is set to the forwarding ip.
For example: TRUSTED_PROXIES=*
Note that the wildcard value (*) is a very insecure option and not recommended.
if you are working behind an apache reverse proxy, that the forwarding headers are properly set.
For example: RequestHeader set X-Forwarded-Proto https is set.
if the CSP is blocking your assets because serving http, you need to force https.
This is done by setting APP_FORCE_HTTPS=true"

![image](https://github.com/LycheeOrg/Lychee-Docker/assets/5970551/6fb8586d-07b2-4143-8ea7-8a94a8c5d175)

I tried to check/change settings mentioned in this message but nothing changed.
I compared docker compose settings for this gallery with my other gallery that is still working and i don't see anything different.

Googling the problem didn't help as it searching a lot of other galleries with the same problem :)

Lychee Version: 5.1.2 (release)
Lychee Commit:  067ac87
https://github.com/LycheeOrg/Lychee/commit/067ac87d5aa8bd4d403fe77e9b0271b8acf789ae

**Comment by Tiritibambix** [comment]
Same problem here

**Comment by sloniupl** [comment]
Unfortunately i didn't found a solution and it is now happening on all of my Lyche instances. Is there anything we can do about it?

**Comment by SKR-974** [comment]
Same problem after updating version 4.11.0 at 5.3.0

**Comment by d7415** [comment]
I'm going to move this to the Lychee repo, as it doesn't look Docker-specific.

We will need more information to debug though. Particular examples:

- Are you hosting Lychee at a subpath?
- Have you tried connecting directly (without any reverse proxy)
- Is your APP_URL set correctly?

**Comment by sloniupl** [comment]
I'm hosting lychee as subdomain and also as a direct domain. 
Without reverse proxy it was showing up partially (interface was loading but no content etc) as the APP_URL was configured correctly for the domain that was in reverse proxy.

Don't know what changed recently but all my galleries are working again correctly.
I have auto-update enabled on my docker images so something changed and it's working correctly now :)

**Comment by SKR-974** [comment]
I can access the site only by being on the local network by using an address 192.168.x.x: 8080,
There he displays the presentation page correctly with my image in the background,
But when I entered the gallery, I only have black squares with the name of the photos,
And from the external address through the reverse proxy I have the message "If you can read me, it means that you misconfigured Lychee."

I have never encountered a problem with the old updates until the 4.11.0,
The problem occurred from 4.11.0 to 5.3.0,

**Reply by ildyria** [reply]
Hi, could you give us a look at your diagnostic page?

**Comment by seth168hh** [comment]
Also same for me. Lychee is working fine in my local network using private IPs, but through nginx Proxy Manager I got the same error message as shown in the initial post.

diagnose shows follows:

Diagnostics
    -----------
        Warning: Dropbox import not working. dropbox_key is empty.
        Warning: You may experience problems when uploading a photo of large size. Take a look in the FAQ for details.
    
    System Information
    ------------------
        Lychee Version (release):                5.3.0
        DB Version:                              5.3.0
        
        composer install:                        --no-dev
        APP_ENV:                                 production
        APP_DEBUG:                               false
        APP_URL:                                 set
        APP_DIR:                                 default
        LOG_VIEWER_ENABLED:                      true
        LIVEWIRE_ENABLED:                        true
        
        System:                                  Linux
        PHP Version:                             8.3.7
        PHP User agent:                          Lychee/4 (https://lycheeorg.github.io/)
        Timezone:                                Europe/Berlin
        Max uploaded file size:                  20M
        Max post size:                           200M
        Livewire chunk size:                     12.00 MB
        Max execution time:                      200
        MySQL Version:                           11.3.2-MariaDB-1:11.3.2+maria~ubu2204
        
        exec() Available:                        yes
        Imagick Available:                       1
        Imagick Enabled:                         1
        Imagick Version:                         1809
        GD Version:                              bundled (2.1.0 compatible)
        Number of foreign key:                   12 found.
    
    Config Information
    ------------------
        version:                                 050300
        check_for_updates:                       0
        sorting_photos_col:                      taken_at
        sorting_photos_order:                    ASC
        sorting_albums_col:                      max_taken_at
        sorting_albums_order:                    ASC
        imagick:                                 1
        skip_duplicates:                         0
        small_max_width:                         0
        small_max_height:                        360
        medium_max_width:                        1920
        medium_max_height:                       1080
        lang:                                    de
        image_overlay_type:                      desc
        default_license:                         CC0
        compression_quality:                     90
        grants_full_photo_access:                1
        delete_imported:                         0
        mod_frame_enabled:                       1
        mod_frame_refresh:                       30
        thumb_2x:                                1
        small_2x:                                1
        medium_2x:                               1
        landing_page_enable:                     0
        force_migration_in_production:           0
        search_public:                           0
        hide_version_number:                     0
        SL_enable:                               0
        SL_for_admin:                            0
        recent_age:                              1
        grants_download:                         0
        photos_wraparound:                       1
        map_display:                             0
        zip64:                                   1
        map_display_public:                      0
        map_provider:                            Wikimedia
        force_32bit_ids:                         0
        map_include_subalbums:                   0
        update_check_every_days:                 3
        has_exiftool:                            1
        share_button_visible:                    0
        import_via_symlink:                      0
        has_ffmpeg:                              1
        location_decoding:                       0
        location_decoding_timeout:               30
        location_show:                           1
        location_show_public:                    0
        rss_enable:                              0
        rss_recent_days:                         7
        rss_max_items:                           100
        prefer_available_xmp_metadata:           0
        editor_enabled:                          1
        lossless_optimization:                   0
        swipe_tolerance_x:                       150
        swipe_tolerance_y:                       250
        local_takestamp_video_formats:           .avi|.mov
        log_max_num_line:                        1000
        unlock_password_photos_with_url_param:   0
        nsfw_visible:                            1
        nsfw_blur:                               0
        nsfw_warning:                            0
        nsfw_warning_admin:                      0
        nsfw_banner_override:                    
        map_display_direction:                   1
        album_subtitle_type:                     oldstyle
        upload_processing_limit:                 4
        new_photos_notification:                 0
        legacy_id_redirection:                   1
        zip_deflate_level:                       6
        SA_enabled:                              1
        default_album_protection:                1
        album_decoration:                        layers
        album_decoration_orientation:            row
        allow_username_change:                   1
        auto_fix_orientation:                    1
        use_job_queues:                          0
        random_album_id:                         starred
        use_last_modified_date_when_no_exif_date: 0
        layout:                                  justified
        date_format_photo_thumb:                 M j, Y, g:i:s A e
        date_format_photo_overlay:               M j, Y, g:i:s A e
        date_format_sidebar_uploaded:            M j, Y, g:i:s A e
        date_format_sidebar_taken_at:            M j, Y, g:i:s A e
        date_format_hero_min_max:                F Y
        date_format_hero_created_at:             M j, Y, g:i:s A T
        date_format_album_thumb:                 M Y
        upload_chunk_size:                       0
        nsfw_banner_blur_backdrop:               0
        search_pagination_limit:                 1000
        search_minimum_length_required:          4
        photo_layout_justified_row_height:       320
        photo_layout_masonry_column_width:       300
        photo_layout_grid_column_width:          250
        photo_layout_square_column_width:        200
        photo_layout_gap:                        12
        display_thumb_album_overlay:             always
        display_thumb_photo_overlay:             hover
        default_album_thumb_aspect_ratio:        1/1
        use_album_compact_header:                0
        login_button_position:                   left
        back_button_enabled:                     0
        back_button_text:                        Return to Home
        current_job_processing_visible:          1

**Reply by ildyria** [reply]
> nginx Proxy Manager

You will need to set some of your header properties in your nginx proxy.
Something like:
```
    proxy_set_header X-Forwarded-Proto https;
```

**Reply by seth168hh** [reply]
Where and how exactly should I do this? I tried it here, but result is still the same:

![Bild_2024-05-26_200255815](https://github.com/LycheeOrg/Lychee/assets/170883252/8333023b-d198-4a88-9e6c-f62e100d4237)

**Reply by ildyria** [reply]
I really don't know Nginx proxy manger, I just use Apache and traefik.
Maybe some of those can help you:
https://github.com/LycheeOrg/Lychee/issues/1855#issuecomment-1557615507

**Reply by seth168hh** [reply]
Thank you for your help, but this made also no difference. 
I found another possible solution on a German Blog to put in this values into the "Custom Ngnix Configuration":

proxy_read_timeout 300;
proxy_pass_header  X-Transmission-Session-Id;
proxy_set_header   X-Forwarded-Host   $host;
proxy_set_header   X-Forwarded-Server $host;
proxy_set_header   X-Forwarded-For    $proxy_add_x_forwarded_for;   

but to get it work, you need to tweak also apache2 with "a2enmod remoteip" mod and this config:

<virtualhost>
   ###
   ###
  RemoteIPHeader X-Forwarded-For
  RemoteIPInternalProxy IP-OF-YOUR-NGINX-MANAGER
  SetEnvIf X-Forwarded-Proto "^https$" HTTPS=on
   ###
   ###
</virtualhost>

and restart: "service apache2 restart"

I connected to my lychee docker container and I did not found Apache2 neither does this container knows "service".

**Reply by ildyria** [reply]
> I connected to my lychee docker container and I did not found Apache2 neither does this container knows "service".

Lychee-docker is using Nginx under the hood. If you want to configure things you will need to edit https://github.com/LycheeOrg/Lychee-Docker/blob/master/default.conf

**Reply by d7415** [reply]
Start with the outside nginx - the internal one should generally be ok.

**Comment by larsburkhardt** [comment]
Any solution for this? Why is it closed?
I just installed 6.60 on my Homeserver and I am behind a Proxy (NGINX Reverse Proxy Manager) -I can't create my Admin user (form requests are http, styles and js don't load

**Reply by ildyria** [reply]
It is closed because most of the time this is configuration problem and we do not have the expertise to debug some particular user setup. We already provide guides to install lychee behind caddy or traefik or Apache reverse proxy. We are not able to cover all the possible combinations of solutions. We provide docker-compose file to quickly get started that should be enough for 99% of our users: 3.4 millions docker pulls vs. 25 000 downloads.

Now to look back at your issue, what happens when you look at the network console of your web browser ?
What kind of errors are your facing?

**Reply by larsburkhardt** [reply, marked as answer]
nvm - managed to get it working. Followed [this comment](https://github.com/LycheeOrg/Lychee/discussions/2424#discussioncomment-9563397):

First it didn't work for me, but a day later and a restart I could access Lychee :)

For questions in the future here my setup

I have a Lychee instance with Docker on a local server in my homelab, public access through a Wireguard tunnel and NGINX Proxy Manager as Reverse Proxy. I don't have to deal with Apache, the settings in the advanced settings of NGINX Proxy Manager:

proxy_read_timeout 300;
proxy_pass_header  X-Transmission-Session-Id;
proxy_set_header   X-Forwarded-Host   $host;
proxy_set_header   X-Forwarded-Server $host;
proxy_set_header   X-Forwarded-For    $proxy_add_x_forwarded_for;   

did the trick for me.
